Makefile

```diff
@@ -50,93 +38,106 @@ endif
 
-ifdef REPO_DIR
-DOCKER_IN_DOCKER=1
-else
-REPO_DIR:=$(shell pwd)
-endif
 
-# You can set this variable for testing and the built image will also be tagged with this name
-OVERRIDE_IMAGE_NAME?=
-
-# If this session isn't interactive, then we don't want to allocate a
-# TTY, which would fail, but if it is interactive, we do want to attach
-# so that the user can send e.g. ^C through.
-INTERACTIVE := $(shell [ -t 0 ] && echo 1 || echo 0)
-TTY=
-ifeq ($(INTERACTIVE), 1)
-	TTY=-t
-endif
+# Rules
+# =====
 
-LDFLAGS=-w -X github.com/kubernetes-incubator/metrics-server/pkg/version.gitVersion=$(GIT_VERSION) -X github.com/kubernetes-incubator/metrics-server/pkg/version.gitCommit=$(GIT_COMMIT) -X github.com/kubernetes-incubator/metrics-server/pkg/version.gitTreeState=$(GIT_TREE_STATE) -X github.com/kubernetes-incubator/metrics-server/pkg/version.buildDate=$(BUILD_DATE)
+.PHONY: all test-unit container container-* clean container-only container-only-* tmp-dir push do-push-* sub-push-*
 
-version-info:
-	@echo "Version: $(GIT_VERSION) ($(VERSION))"
-	@echo "    built from $(GIT_COMMIT) ($(GIT_TREE_STATE))"
-	@echo "    built on $(BUILD_DATE)"
+# Build Rules
+# -----------
 
-fmt:
-	find . -type f -name "*.go" | grep -v "./vendor*" | xargs gofmt -s -w
+# building depends on all go files (this is mostly redundant in the face of go 1.10's incremental builds,
+# but it allows us to safely write actual dependency rules in our makefile)
+src_deps=$(shell find pkg cmd -type f -name "*.go")
+_output/%/metrics-server: $(src_deps)
+	GOARCH=$* CGO_ENABLED=0 go build -ldflags "$(LDFLAGS)" -o _output/$*/metrics-server github.com/kubernetes-incubator/metrics-server/cmd/metrics-server
 
-build: fmt
-	GOARCH=$(ARCH) CGO_ENABLED=0 go build -ldflags "$(LDFLAGS)" -o _output/$(ARCH)/metrics-server github.com/kubernetes-incubator/metrics-server/cmd/metrics-server
+# Image Rules
+# -----------
 
-test-unit:
-ifeq ($(ARCH),amd64)
-	GOARCH=$(ARCH) go test --test.short -race ./pkg/... $(FLAGS)
-else
-	GOARCH=$(ARCH) go test --test.short ./pkg/... $(FLAGS)
-endif
+# build a container using containerized build (the current arch by default)
+container: container-$(ARCH)
 
-container:
+container-%: tmpdir
 	# Run the build in a container in order to have reproducible builds
-	# Also, fetch the latest ca certificates
-	docker run --rm -i $(TTY) -v $(TEMP_DIR):/build -v $(REPO_DIR):/go/src/github.com/kubernetes-incubator/metrics-server -w /go/src/github.com/kubernetes-incubator/metrics-server golang:$(GOLANG_VERSION) /bin/bash -c "\
-		cp /etc/ssl/certs/ca-certificates.crt /build \
+	docker run --rm -v $(TEMP_DIR):/build -v $(REPO_DIR):/go/src/github.com/kubernetes-incubator/metrics-server -w /go/src/github.com/kubernetes-incubator/metrics-server golang:$(GOLANG_VERSION) /bin/bash -c "\
 		&& GOARCH=$(ARCH) CGO_ENABLED=0 go build -ldflags \"$(LDFLAGS)\" -o /build/metrics-server github.com/kubernetes-incubator/metrics-server/cmd/metrics-server"
```

Comment:
You dropped the `cp` command above, now the first line starts with `&&` and this breaks the `container` target in the Makefile